wsclient/wsclient_test.go

```diff
@@ -592,3 +613,78 @@ func (handler *testHandler) ProcessMessage(
 
 func (handler *testHandler) ClientDisconnected(client *wsserver.Client) {
 }
+
+func savePEMFile(data []byte) (string, error) {
+	file, err := ioutil.TempFile(tmpDir, "*."+cryptutils.PEMExt)
+	if err != nil {
+		return "", aoserrors.Wrap(err)
+	}
+	defer file.Close()
+
+	if _, err = file.Write(data); err != nil {
+		return "", aoserrors.Wrap(err)
+	}
+
+	return file.Name(), nil
+}
+
+func prepareTestCert() error {
+	certCA, key, err := testtools.GenDefaultCARootCertPEMAndKey()
+	if err != nil {
+		return aoserrors.Wrap(err)
+	}
+
+	subject := testtools.DefaultCertificateTemplate.Subject
+	subject.CommonName = "AoS Secondary CA"
+
+	certSecond, keySecond, err := testtools.GenCACertPEMAndKey(certCA, key, subject)
+	if err != nil {
+		return aoserrors.Wrap(err)
+	}
+
+	subject = testtools.DefaultCertificateTemplate.Subject
+	subject.OrganizationalUnit = []string{"Novus Ordo Seclorum"}
+	subject.CommonName = "AOS vehicles Intermediate CA"
+
+	certInter, keyInter, err := testtools.GenCACertPEMAndKey(certSecond, keySecond, subject)
+	if err != nil {
+		return aoserrors.Wrap(err)
+	}
+
+	subject = testtools.DefaultCertificateTemplate.Subject
+	subject.CommonName = "AoS update manager"
+
+	cert, keyUpdateManager, err := testtools.GenCertPEMAndKey(subject, certInter, keyInter)
+	if err != nil {
+		return aoserrors.Wrap(err)
+	}
+
+	var certChain []byte
+
+	certChain = append(certChain, cryptutils.CertToPEM(cert)...)
```

Comment:
As I could understand this triple certificate will be used often? Would it be good to create a function that generates them instead of 3 separate functions?

testtools.GenCACertAndKey() ([]*x509.Certificate, crypto.PrivateKey, error)

Reply:
Decided to leave it as is.